bpo-37599: Remove a vague statement in documentation of Integer Objects

sgalal opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->

In current Python document (3.7 - 3.9) there is such statement in documentation of Integer Objects:

> The current implementation keeps an array of integer objects for all integers between ``-5`` and ``256``, when you create an int in that range you actually just get back a reference to the existing object. So it should be possible to change the value of ``1``.  I suspect the behaviour of Python in this case is undefined. :-)

The last sentence is vague. It is inappropriate to write "I suspect" in documentation. And as the statements ahead has already clarified the sense that this is "a current implementation", **the last sentence should be removed**.

https://bugs.python.org/issue37599

<!-- issue-number: [bpo-37599](https://bugs.python.org/issue37599) -->
https://bugs.python.org/issue37599
<!-- /issue-number -->

---

Edit: Change "irresponsible" to "inappropriate" to avoid pejoratives, according to <https://bugs.python.org/msg348019>.

sgalal committed:
Remove a vague statement in documentation

aeros commented on Doc/c-api/long.rst:
```diff
@@ -43,8 +43,7 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
    The current implementation keeps an array of integer objects for all integers
    between ``-5`` and ``256``, when you create an int in that range you actually
    just get back a reference to the existing object. So it should be possible to
```
```suggestion
    just get back a reference to the existing object.
```
Edit: Removal of the next line is included in a comment below

aeros reviewed (approved):
@sgalal Welcome and thanks for the contribution! 

I completely agree with the removal of the vague statement. While cleaning up the summary for this function, I would also recommend removing the previous sentence as well. It is also quite vague and doesn't add significant value. A sentence starting with "So it should be possible..." shouldn't be in the docs either.

aeros commented on Doc/c-api/long.rst:
```diff
@@ -43,8 +43,7 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
    just get back a reference to the existing object. So it should be possible to
-   change the value of ``1``.  I suspect the behaviour of Python in this case is
-   undefined. :-)
+   change the value of ``1``.
```
Meant to include the removal of this line in my previous review, part of the previous sentence I was referring to:
```suggestion
```

sgalal committed:
Remove another vague sentence

A sentence starting with "So it should be possible..." shouldn't be in the docs either.

Co-Authored-By: Kyle Stanley <aeros167@gmail.com>

sgalal committed:
Include the removal of the previous line

Co-Authored-By: Kyle Stanley <aeros167@gmail.com>

sgalal committed:
Remove an extra space
